sdk/formrecognizer/ai-form-recognizer/src/models.ts

```diff
@@ -256,78 +256,201 @@ export interface FieldData {
+ *
+ * This interface is provided for type safety and should only be encountered
+ * when the `valueType` of a FormField is undefined, and there is no ordinary
+ * reason that should be the case.
+ */
+export interface FormUnknownField extends FormFieldCommon {
+  valueType?: undefined;
+  /**
+   * If `valueType` is undefined, then the type of the value is unknown.
+   */
+  value?: unknown;
+}
+
+/**
+ * A form field with a string value.
+ */
+export interface FormStringField extends FormFieldCommon {
+  valueType: "string";
+  /**
+   * The value of the recognized string.
+   */
+  value?: string;
+}
+
+/**
+ * A form field with a numeric value.
+ */
+export interface FormNumberField extends FormFieldCommon {
+  valueType: "number";
+  /**
+   * The value of the recognized number.
+   */
+  value?: number;
+}
+
+/**
+ * A form field with a value representing a date.
+ */
+export interface FormDateField extends FormFieldCommon {
+  valueType: "date";
+  /**
+   * The value of the date field, represented as a JavaScript date object.
+   */
+  value?: Date;
+}
+
+/**
+ * A form field with a value representing a time.
+ */
+export interface FormTimeField extends FormFieldCommon {
+  valueType: "time";
+  /**
+   * The value of the time field, represented as a string.
+   */
+  value?: string;
+}
+
+/**
+ * A form field with a value representing a phone number.
+ */
+export interface FormPhoneNumberField extends FormFieldCommon {
+  valueType: "phoneNumber";
+  /**
+   * The value of the recognized phone number, represented as a string.
+   *
+   * This value is normalized to a uniform string representation, e.g.
+   * "+447911123456". If the service is unable to normalize the value,
+   * information about the phone number's textual appearance may appear in the
+   * `text` property of the value data (see the `valueData` field of this
+   * object).
+   */
+  value?: string;
+}
+
+/**
+ * A form field with an integer value.
+ */
+export interface FormIntegerField extends FormFieldCommon {
+  valueType: "integer";
+  /**
+   * The value of the recognized integer.
+   */
+  value?: number;
+}
+
+/**
+ * A form field with an array of FormFields as a value.
+ */
+export interface FormArrayField extends FormFieldCommon {
+  valueType: "array";
+  /**
+   * The recognized array of nested fields. Each value in this array is its
+   * own `FormField`.
+   */
+  value?: FormField[];
+}
+
+/**
+ * A form field with a key-value map (an "object") as a value.
+ */
+export interface FormObjectField extends FormFieldCommon {
+  valueType: "object";
+  /**
+   * The recognized object structure of the field, represented as a JavaScript
+   * object with the nested fields' names as properties that store their
+   * values.
+   */
+  value?: Record<string, FormField>;
+}
+
+/**
+ * A form field with a value representing the state of a selection mark.
+ */
+export interface FormSelectionMarkField extends FormFieldCommon {
```

Comment:
idk if it will matter too much, but because the actual value is a `SelectionMarkState` consider if a rename to `formSelectionMarkStateField` is necessary

Reply:
IMO the FormField represents more than just the state of the selection mark, as it also provides information such as label and value textual information, bounding boxes, etc. In any case, I like that these types line up perfectly with the values of `valueType`, and changing to FormSelectionMarkStateField would break that.

Reply:
Ahh I see. I missed that it extends from `FormFieldCommon` in which case you are right.
I got confused because in .NET we only return the type, not the whole `FormField`